spec/cloud.go

```diff
@@ -314,9 +320,20 @@ func (g gceMeta) toGeneratorResults() *mackerel.Cloud {
 	return &mackerel.Cloud{Provider: "gce", MetaData: g.toGeneratorMeta()}
 }
 
-// SuggestCustomIdentifier for GCE is not implemented yet
+func builtGCECustomIdentifier(instanceID string) string {
+	return instanceID + ".gce.cloud.google.com"
+}
+
+// SuggestCustomIdentifier the identifier of the GCE VM instance
 func (g *GCEGenerator) SuggestCustomIdentifier() (string, error) {
-	return "", nil
+	if g.meta.Instance == nil || g.meta.Instance.InstanceID == 0 {
```

Comment:
Should we consider the case `g.meta == nil`?

Reply:
We don't have to check `g.meta == nil` case because g.meta is not reference type. So always g.meta have value even if Generate() was not called.